src/hooks/useClaimOrderCallback.ts

```diff
@@ -140,57 +152,49 @@ export function useGetAuctionProceeds(
   }
 }
 
-export function useClaimOrderCallback(
+export const useClaimOrderCallback = (
   auctionIdentifier: AuctionIdentifier,
-): null | (() => Promise<string>) {
+): (() => Promise<Maybe<string>>) => {
   const { account, library } = useActiveWeb3React()
   const addTransaction = useTransactionAdder()
 
   const { auctionId, chainId } = auctionIdentifier
-  const claimInfo = useGetClaimInfo(auctionIdentifier)
+  const { claimInfo, error } = useGetClaimInfo(auctionIdentifier)
   const gasPrice = useGasPrice(chainId)
 
-  return useMemo(() => {
-    return async function onClaimOrder() {
-      if (!chainId || !library || !account || !claimInfo) {
-        throw new Error('missing dependencies in onPlaceOrder callback')
-      }
-
-      const easyAuctionContract: Contract = getEasyAuctionContract(
-        chainId as ChainId,
-        library,
-        account,
-      )
-
-      let estimate,
-        method: Function,
-        args: Array<string | string[] | number>,
-        value: Maybe<BigNumber>
-      {
-        estimate = easyAuctionContract.estimateGas.claimFromParticipantOrder
-        method = easyAuctionContract.claimFromParticipantOrder
-        args = [auctionId, claimInfo?.sellOrdersFormUser]
-        value = null
-      }
+  const claimCallback = useCallback(async (): Promise<Maybe<string>> => {
+    if (!chainId || !library || !account || error || !claimInfo) {
+      // throw new Error('missing dependencies in onPlaceOrder callback')
```

Comment:
I think throwing an error is okay, this makes the application easier to debug, doesn't. it?

Reply:
Added again, and removed the try catch at the end of the block, to throw an exception , so that the entire block behaves in the same manner